odoo/osv/expression.py

```diff
@@ -710,9 +711,7 @@ def push_result(query, params):
                 # Non-stored field should provide an implementation of search.
                 if not field.search:
                     # field does not support search!
-                    _logger.error("Non-stored field %s cannot be searched.", field)
-                    if _logger.isEnabledFor(logging.DEBUG):
-                        _logger.debug(''.join(traceback.format_stack()))
+                    _logger.error("Non-stored field %s cannot be searched.", field, exc_info=sys.exc_info())
```

Comment:
We are not in an exception handler, so `sys.exc_info()` will return None, or at worst an unrelated exception. There is the `stack_info` argument for the current traceback.
```suggestion
                    _logger.error("Non-stored field %s cannot be searched.", field, stack_info=True)
```

Reply:
Unfortunately, `sentry` doesn't respect it.
So using `exc_info=True` should be used.

And the `if _logger.isEnabledFor(logging.DEBUG):` block should be kept.